File: app/src/main/java/com/example/sarahn/waterreminderapp/fragments/SettingsFragment.java
package com.example.sarahn.waterreminderapp.fragments;

import android.app.Activity;
import android.graphics.Color;
import android.net.Uri;
import android.os.Bundle;

import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.sarahn.waterreminderapp.R;
import com.example.sarahn.waterreminderapp.Utils.NotificationUtills;
import com.example.sarahn.waterreminderapp.Utils.SharedPrefUtils;
import com.example.sarahn.waterreminderapp.Utils.ToastLogger;
import com.example.sarahn.waterreminderapp.classes.ClimateBottomSheet;
import com.example.sarahn.waterreminderapp.classes.ClsAMOrPM;
import com.example.sarahn.waterreminderapp.classes.ClsRequirementCalculator;
import com.example.sarahn.waterreminderapp.classes.MyApplication;
import com.example.sarahn.waterreminderapp.dialogs.ClsTimePickerDialogBuilder;
import com.example.sarahn.waterreminderapp.customsViews.CustomBtnFonty;
import com.example.sarahn.waterreminderapp.settings.ClsBottomSheetActivity;
import com.example.sarahn.waterreminderapp.settings.ClsBottomSheetWeight;

public class SettingsFragment extends Fragment  implements OnClickListener {

    private OnFragmentInteractionListener mListener;

    public static TextView tvOne;
    public static TextView tvTwo;
    public static TextView tvThree;
    public static TextView tvSpan;
    public static CustomBtnFonty btnUpdate;
    public static boolean isFromSettings;




    public SettingsFragment() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        View view = inflater.inflate(R.layout.fragment_settings, container, false);


        tvOne = (TextView) view.findViewById(R.id.tv_one);
        tvTwo = (TextView) view.findViewById(R.id.tv_two);
        tvThree = (TextView) view.findViewById(R.id.tv_three);
        tvSpan = (TextView) view.findViewById(R.id.tv_span);
        btnUpdate = (CustomBtnFonty) view.findViewById(R.id.recalculate);

        tvOne.setText(Integer.toString(SharedPrefUtils.getWeight(getContext())));
        tvTwo.setText(SharedPrefUtils.getActivityLevel(getContext()));
        tvThree.setText(SharedPrefUtils.getClimate(getContext()));
        btnUpdate.setBackgroundColor(Color.parseColor("#ff0000"));

        tvSpan.setText(SharedPrefUtils.getStartHour(getContext())+":"
        + SharedPrefUtils.getStartMin(getContext())+ "-" + SharedPrefUtils.getEndHour(getContext())
        +":" + SharedPrefUtils.getEndMin(getContext()));


        tvOne.setOnClickListener(this);
        tvTwo.setOnClickListener(this);
        tvThree.setOnClickListener(this);
        tvSpan.setOnClickListener(this);
        btnUpdate.setOnClickListener(this);

        return view;
    }


    public void onButtonPressed(Uri uri) {
        if (mListener != null) {
            mListener.onFragmentInteraction(uri);
        }
    }

    @Override
    public void onAttach(Activity activity) {
        super.onAttach(activity);
        try {
            mListener = (OnFragmentInteractionListener) activity;
        } catch (ClassCastException e) {
            throw new ClassCastException(activity.toString()
                    + " must implement OnFragmentInteractionListener");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }

    @Override
    public void onClick(View v) {

        switch (v.getId()){
            case R.id.tv_one:
                ClsBottomSheetWeight bottomSheet = new ClsBottomSheetWeight(getContext());
                bottomSheet.showDialog().show();
                //make the bellow line of code only works when it return from bottomsheet
                // tvOne.setText(Integer.toString(SharedPrefUtils.getWeight(this)));
                break;

            case R.id.tv_two:
                ClsBottomSheetActivity activityBottomSheet = new ClsBottomSheetActivity(getContext());
                activityBottomSheet.showDialog().show();
                break;

            case R.id.tv_three:
                ClimateBottomSheet bottomSheetClimate = new ClimateBottomSheet(getContext());
                bottomSheetClimate.showDialog().show();
                break;

            case R.id.tv_span:
                isFromSettings = true;
                ClsTimePickerDialogBuilder.showDialog(getContext()).show(getActivity().getFragmentManager(), "timepicker");
                break;
            case R.id.recalculate:
                ClsRequirementCalculator.calculateRequirement(getContext());
                NotificationUtills.NotificationCounter();
                ToastLogger.toastMessage("Updated");
                break;
        }

    }


    public interface OnFragmentInteractionListener {
        
        public void onFragmentInteraction(Uri uri);
    }

}

File: app/src/main/java/com/example/sarahn/waterreminderapp/fragments/SettingsFragment.kt
package com.example.sarahn.waterreminderapp.fragments

import android.app.Activity
import android.graphics.Color
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.example.sarahn.waterreminderapp.R
import com.example.sarahn.waterreminderapp.Utils.NotificationUtills
import com.example.sarahn.waterreminderapp.Utils.SharedPrefUtils
import com.example.sarahn.waterreminderapp.Utils.ToastLogger
import com.example.sarahn.waterreminderapp.classes.*
import com.example.sarahn.waterreminderapp.classes.ClsRequirementCalculator.calculateRequirement
import com.example.sarahn.waterreminderapp.customsViews.CustomBtnFonty
import com.example.sarahn.waterreminderapp.dialogs.ClsTimePickerDialogBuilder.showDialog

class SettingsFragment : Fragment(), View.OnClickListener {
    private var mListener: OnFragmentInteractionListener? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_settings, container, false)
        tvOne = view.findViewById<View>(R.id.tv_one) as TextView
        tvTwo = view.findViewById<View>(R.id.tv_two) as TextView
        tvThree = view.findViewById<View>(R.id.tv_three) as TextView
        tvSpan = view.findViewById<View>(R.id.tv_span) as TextView
        btnUpdate = view.findViewById<View>(R.id.recalculate) as CustomBtnFonty
        tvOne!!.text = Integer.toString(SharedPrefUtils.getWeight(context))
        tvTwo!!.text = SharedPrefUtils.getActivityLevel(context)
        tvThree!!.text = SharedPrefUtils.getClimate(context)
        btnUpdate!!.setBackgroundColor(Color.parseColor("#ff0000"))
        tvSpan!!.text = (SharedPrefUtils.getStartHour(MyApplication.context).toString() + ":"
                + SharedPrefUtils.getStartMin(MyApplication.context) + "-" + SharedPrefUtils.getEndHour(MyApplication.context)
                + ":" + SharedPrefUtils.getEndMin(MyApplication.context))
        tvOne!!.setOnClickListener(this)
        tvTwo!!.setOnClickListener(this)
        tvThree!!.setOnClickListener(this)
        tvSpan!!.setOnClickListener(this)
        btnUpdate!!.setOnClickListener(this)
        return view
    }

    fun onButtonPressed(uri: Uri?) {
        if (mListener != null) {
            mListener!!.onFragmentInteraction(uri)
        }
    }

    override fun onAttach(activity: Activity) {
        super.onAttach(activity)
        mListener = try {
            activity as OnFragmentInteractionListener
        } catch (e: ClassCastException) {
            throw ClassCastException(activity.toString()
                    + " must implement OnFragmentInteractionListener")
        }
    }

    override fun onDetach() {
        super.onDetach()
        mListener = null
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.tv_one -> {
                isFromSettigBottomSheet = true
                val bottomSheet = ClsBottomSheetWeight(requireContext())
                bottomSheet.showDialog().show()
            }
            R.id.tv_two -> {
                isFromSettigBottomSheet = true
                val activityBottomSheet = ClsBottomSheetActivity(requireContext())
                activityBottomSheet.showDialog().show()
            }
            R.id.tv_three -> {
                isFromSettigBottomSheet = true
                val bottomSheetClimate = ClimateBottomSheet(requireContext())
                bottomSheetClimate.showDialog().show()
            }
            R.id.tv_span -> {
                isFromSettings = true
                showDialog(context).show(activity?.fragmentManager, "timepicker")
            }
            R.id.recalculate -> {
                calculateRequirement(context)
                NotificationUtills.NotificationCounter()
                ToastLogger.toastMessage("Updated! new duration " + SharedPrefUtils.getDuration(context))
                ToastLogger.toastMessage("Will be applicable on next span")
            }
        }
    }

    interface OnFragmentInteractionListener {
        fun onFragmentInteraction(uri: Uri?)
    }

    companion object {
        var isFromSettings = false
        @JvmField
        var tvOne: TextView? = null
        @JvmField
        var tvTwo: TextView? = null
        @JvmField
        var tvThree: TextView? = null
        var tvSpan: TextView? = null
        var btnUpdate: CustomBtnFonty? = null
        var isFromSettigBottomSheet = false
    }
}